frontend/public/components/cluster-settings/oauth.tsx

```diff
@@ -73,17 +64,43 @@ const OAuthDetails: React.FC<OAuthDetailsProps> = ({ obj }: { obj: OAuthKind })
   const { identityProviders, tokenConfig } = obj.spec;
   const { t } = useTranslation();
 
-  const IDPDropdownItems = _.keys(addIDPItems).map((idp) => {
-    const label = t('public~{{label}}', { label: addIDPItems[idp] });
+  const getAddIDPItemLabels = (type: string) => {
+    switch (type) {
+      case 'Basic Authentication':
+        return t('public~Basic Authentication');
+      case 'GitHub':
+        return t('public~GitHub');
+      case 'GitLab':
+        return t('public~GitLab');
+      case 'Google':
+        return t('public~Google');
+      case 'HTPasswd':
+        return t('public~HTPasswd');
+      case 'Keystone':
+        return t('public~Keystone');
+      case 'LDAP':
+        return t('public~LDAP');
+      case 'OpenID Connect':
+        return t('public~OpenID Connect');
+      case 'Request Header':
+        return t('public~Request Header');
+      default:
+        return type;
+    }
+  };
+
+  const IDPDropdownItems = Object.entries(addIDPItems).map((idp) => {
+    const [key, value] = idp;
+
     return (
       <DropdownItem
-        key={`idp-${addIDPItems[idp]}`}
```

Comment:
@rhamilto @rebeccaalpert I am not sure why we were assigning `addIDPItems` object value here. I think it should be the key.